url/src/comp.php

```diff
@@ -220,156 +225,308 @@ function counter_summary() {
+				$new_web_address_array = check_web_addresses($oldURL, $newURL, $web_address_array, "replace");
+				$report->newWebAddressArray = $new_web_address_array;
+
+				$new_online_resource = $newURL;
+
+				// Nothing further to do if there are no changes to make
+				if($web_address_array === $new_web_address_array && $online_resource === $new_online_resource){
+					echo_message_to_screen(DEBUG, "Nothing to update with this resource</br>");
+					$report->actionMessage = "Nothing to do";
+					continue;
+				}
+				// build the PATCH body
+				$body = build_patch_body($resource_id, $new_web_address_array, $new_online_resource);
+				echo_message_to_screen(DEBUG, "replace_url patch request body: $body");
+				// if not a dry run - update
+				if ($shouldWritetoLive === "true") {
+					$result = post_url($shortCode, $resource_id, $body, $TalisGUID, $token);
+					if ($result) {
+						$report->actionMessage = "URL replaced successfully";
+						$report->updated = true;
+						$any_resource_updated = true;
+					} else {
+						$report->actionMessage = "URL not replaced";
+						$report->failure = true;
+						$any_resource_updated = false;
+					}
+				} else {
+					$itemReport->actionMessage = "Dry Run";
+					$report->actionMessage = "Would update";
+					$report->updated = true;
+				}
+				increment_counter('URLs replaced: ');
 			} else {
-				fwrite($myfile, "Test Run\r\n");
+				$report->actionMessage = "Nothing to do";
+				$report->updated = true;
 			}
-			increment_counter('URLs replaced: ');
+		}
+		// We only want to flag these if they do not happen for any resource in this item
+		if ($web_address_found === false){
+			echo_message_to_screen(INFO, "Web Address not found in any resource\t");
+			$itemReport->currentWebAddressArray = "Web Address not found in any resource";
+		}
+
+		if ($online_resource_found === false){
+			echo_message_to_screen(INFO, "Online Resource is not set\t");
+			$itemReport->currentOnlineResource = "Online Resource is not set";
+		}
+
+		if ($any_resource_updated === true){
+			$itemReport->actionMessage = "Item updated";
 		} else {
-			fwrite($myfile, "Resource URL Not Updated - Dry Run\r\n");
+			$itemReport->actionMessage = "Item not updated";
 		}
 	}
+	return $itemReport;
+}
 
+/**
+ * Extract all resources from the item data returned from the API
+ * There may be 0 or more resources found
+ *
+ * @param array $itemData
+ * @return array
+ */
+function get_resources_from_item(stdClass $itemData){
+	$resources = [];
+	if (!isset($itemData->included)){
+		return $resources;
+	}
+	foreach ($itemData->included as $v){
+		if(!isset($v->type)){
+			continue;
+		}
+		if($v->type == "resources"){
+			$resources[] = $v;
+		}
+	}
+	return $resources;
 }
 
-function get_item($shortCode, $itemID, $TalisGUID, $token) {
-	$item_lookup = "https://rl.talis.com/3/" . $shortCode . "/draft_items/" . $itemID . "?include=resource";
+function get_item($shortCode, $itemID, $TalisGUID, $token, $includePartOf=true) {
```

Comment:
When we add a web address we don't want the part of records because we only want to add the URL to the main resource. i.e. Add the URL to the chapter, or article, and not the book or journal.

The added includePartOf flag defaults to true but we pass a false value in the `add_url()` function to override this.